lib/spack/spack/solver/concretize.lp

```diff
@@ -7,6 +7,47 @@
 % This logic program implements Spack's concretizer
 %=============================================================================
 
+%-----------------------------------------------------------------------------
+% Map literal input specs to facts that drive the solve
+%-----------------------------------------------------------------------------
+
+% Give clingo the choice to solve an input spec or not
+{ literal_solved(ID) } :- literal(ID).
+literal_not_solved(ID) :- not literal_solved(ID), literal(ID).
+
```

Comment:
I think we can constrain this in asp to ensure at least one spec is concretized, that will get us better error messages than doing it in python.

```
:- C = #count{ ID : literal_solved(ID) }, C<1.
```

Reply:
I agree with that since we now have #30669 Good catch, since this was not pointed out by a conflict during rebase!

Reply:
Should be solved in 7c9ab7156fcf6c0ce905ee3f782760a0da0829e3